polus-imagenet-model-featurization-plugin/src/main.py

```diff
@@ -0,0 +1,142 @@
+    if dim == 2:
+        img = np.stack((img,)*3, axis=-1)
+
+    if target_size is not None:
+        img = transform.resize(img, output_shape=target_size, anti_aliasing=True)
+    
+    return img
+
+
+if __name__=='__main__':
+    # Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    logger = logging.getLogger('main')
+    logger.setLevel(logging.INFO)
+
+    ''' Argument parsing '''
+    logger.info('Parsing arguments...')
+    parser = argparse.ArgumentParser(prog='main', description='Image featurization using models pre-trained on ImageNet')
+    
+    # Input arguments
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input image collection to be processed by this plugin', required=True)
+    parser.add_argument('--model', dest='model', type=str,
+                        help='Pre-trained ImageNet model to use for featurization', required=True)
+    parser.add_argument('--resolution', dest='resolution', type=str,
+                        help='Resolution to which the input images are scaled', required=False)
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+
+    input_dir = args.inpDir
+    logger.info('inpDir = {}'.format(input_dir))
+    
+    model = args.model
+    logger.info('model = {}'.format(model))
+
+    resolution = args.resolution
+    logger.info('resolution = {}'.format(resolution))
+
+    output_dir = args.outDir
+    logger.info('outDir = {}'.format(output_dir))
+
+    if not os.path.exists(output_dir):
+        logger.info(f'Output directory {output_dir} does not exist. Creating directory.')
+        os.makedirs(output_dir)
+
+    # Validate resolution.
+    match = re.match('(\d+)x(\d+)', resolution, flags=re.IGNORECASE)
+    if match is None:
+        logger.error(f'Resolution should be in the following format: 500x500. You entered: {resolution}.')
+        sys.exit()
+    
+    target_size = (int(match.group(1)), int(match.group(2)))
+    logger.info(f'Parsed resolution: {target_size[0]}x{target_size[1]}.')
+
+    # Validate model. 
+    if model not in valid_models:
+        logger.error(f'You requested model {model}. Model must be one of the following: {", ".join(valid_models)}.')
+        sys.exit()
+
+    # Retrieve desired model.
+    imagenet_model = get_imagenet_model(model)
+    if not isinstance(imagenet_model, tf.keras.Model):
+        logger.error(f'Unable to load requested model!')
+        sys.exit()
+    
+    # Get all images. 
+    image_files = [os.path.join(dirpath, file)
+                   for dirpath, _, files in os.walk(input_dir)
+                   for file in files 
+                   if os.path.isfile(os.path.join(dirpath, file)) and file.endswith('.ome.tif')]
```

Comment:
Just a few thoughts.

There is no explicit need to use `bfio` for this particular task. However, if/when we switch over to using a different file format such as `ome.zarr`, using `bfio` would make the transition easier. Also, I suspect that `bfio` might increase the read speed of `ome.tif` files.

Reply:
Thanks, I'll switch over to bfio.